### PR TITLE
Add lc-compliance rootfs

### PR DESCRIPTION
**Pull request by nfraprado**
Add a rootfs and accompanying build script that contains libcamera and
its test suite: lc-compliance.

Running lc-compliance will be useful to test for regressions in the
media subsystem.

Signed-off-by: Nícolas F. R. A. Prado <nfraprado@collabora.com>

### CODE REVIEW

File: config/core/rootfs-configs.yaml [hunk 1]
```diff
@@ -1,4 +1,21 @@
...
+      - arm64
+      - armhf
+    extra_packages:
+      - ca-certificates
+      - libevent-dev
+    extra_packages_remove:
+      - bash
+      - e2fslibs
+      - e2fsprogs
+    script: "scripts/bullseye-libcamera.sh"
```
Comment (gctucker) [inline config/core/rootfs-configs.yaml:16]: This looks fine.  If you have any issues with the rootfs, maybe drop the `script` line initially to check whether the basic `bullseye` rootfs image boots successfully.  That would help you determine if any issues are due to the Debian rootfs or the libcamera build script.

File: config/rootfs/debos/scripts/bullseye-libcamera.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+echo '{  "tests_suites": [' >> $BUILDFILE
+
+# Build libcamera and lc-compliance
+########################################################################
+
+LIBCAMERA_URL=https://gitlab.collabora.com/nfraprado/libcamera.git
+mkdir -p /tmp/tests/libcamera && cd /tmp/tests/libcamera
+
+# Use my fork until it gets merged
+git clone --single-branch --branch lcc-gtest-v5 --depth=1 $LIBCAMERA_URL .
```
Comment (alexandrasp) [inline config/rootfs/debos/scripts/bullseye-libcamera.sh:36]: @nfraprado I tried to build a rootfs on staging using your commit and it is breaking looking for git in line 36, so probably you need to install it before using. I am not sure if you have access to jenkins (I don't think so), then I will paste some log lines:

`2021/05/28 00:29:53 bullseye-libcamera.sh | Processing triggers for libc-bin (2.31-12) ...

2021/05/28 00:29:53 bullseye-libcamera.sh | /tmp/script/bullseye-libcamera.sh: line 36: git: command not found
2021/05/28 00:29:53 Action `Build testsuite` failed at stage Run, error: exit status 127`

Comment (nfraprado) [reply config/rootfs/debos/scripts/bullseye-libcamera.sh:39]: Okay, I missed that indeed, thanks for the feedback. I just force-pushed a commit with git included, could you try and see if it works now?

**Comment by alexandrasp** [comment]
@nfraprado arm64 rootfs still missing some detail, but arm and x86-64 are available [here](https://storage.staging.kernelci.org/images/rootfs/debian/bullseye-libcamera/20210531.0/):

**Comment by nfraprado** [comment]
It seems that the arm64 issue is due to a minimum version requirement of QEMU >= 5.0 as described in [this thread](https://linux.debian.bugs.dist.narkive.com/KYbYWWu7/bug-987497-libc-bin-enabling-static-pie-support-on-arm64-libc-breaks-qemu-bootstrapping#post3). For now, I've reworked the rootfs to use buster instead and just fetch libgtest from buster-backports and meson from pip. @alexandrasp could you please re-spin the image? Hopefully it works now.

**Comment by mgalka** [comment]
I'm able to build `amd64`, `armhf` and `arm64` variants with `kernelci/debos` based docker container.
@gctucker @nfraprado Do you think we can mark this PR as "Ready for review"?

**Comment by nfraprado** [comment]
Hi @mgalka , it did compile but the lc-compliance binary wasn't generated since it was missing a build-time dependency (I assumed the packages in `extra_packages` in the yaml were installed before the build script was run, but that isn't the case).

I've now added `libevent-dev` as a dependency in the build script and tested locally and now everything seems good. So I'll mark this as ready for review.

**Comment by nfraprado** [comment]
Added a proper commit message :).

**Comment by alexandrasp** [comment]
@mgalka @gctucker tested OK on staging and rootfs images are available [here](https://storage.staging.kernelci.org/images/rootfs/debian/buster-libcamera/)

**Comment by nfraprado** [comment]
Not sure if it would be best to leave the parser script for a different PR, but I figured since it's part of the rootfs it could be added now as well. It can be seen working [here](https://lava.collabora.co.uk/scheduler/job/3934272).

**Review by gctucker — APPROVED** [review]
@alexandrasp Please take another look now there's a working LAVA job submitted by hand and merge if it's all OK.

**Comment by gctucker** [comment]
The jobs were scheduled but failed on `hana`:
https://staging.kernelci.org/test/plan/id/60b85d0c809f071784902fc8/
https://storage.staging.kernelci.org/mainline/master/v5.13-rc4-61-g324c92e5e0ee/arm64/defconfig/clang-10/lab-collabora/lc-compliance-mt8173-elm-hana.html

It looks like this could be because the ramdisk is too large.

@nfraprado @alexandrasp Can you please take a look and maybe try again with NFS root?

File: config/rootfs/debos/overlays/libcamera/usr/bin/lc-compliance-parser.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Get first camera name
```
Comment (gctucker) [inline config/rootfs/debos/overlays/libcamera/usr/bin/lc-compliance-parser.sh:1]: Missing shebang line.

Comment (nfraprado) [reply config/rootfs/debos/overlays/libcamera/usr/bin/lc-compliance-parser.sh:3]: Sorry. Fixed.

**Comment by nfraprado** [comment]
First of all, I messed up the path of the parser script so I fixed that in the last push. Now, @gctucker regarding the fail you pointed to, we ran these two tests: https://lava.collabora.co.uk/scheduler/job/3956636 and https://lava.collabora.co.uk/scheduler/job/3956675 . The first one uses nfs and succeeds running the tests, while the second uses ramdisk and fails in the same way that one you pointed to.

So I guess the fix is for me to change #720 to use nfs and a initrd instead of just a ramdisk, and using `debian-staging` as the type, so we could close #723, right?

**Comment by gctucker** [comment]
Building a new set of rootfs images now with the latest changes:
https://bot.staging.kernelci.org/job/rootfs-build-trigger/9/console

**Review by alexandrasp — APPROVED** [review]

**Comment by gctucker** [comment]
See comments on #720, this is being tested on staging now.

File: config/rootfs/debos/overlays/libcamera/usr/bin/lc-compliance-parser.sh [hunk 4]
```diff
@@ -33,7 +33,8 @@ lc-compliance -c "$(get_cam_name)" | while read line; do
 
 	# Check for test result
 	echo "$line" | grep -qe '^\[       OK \]' && res=pass
-	echo "$line" | grep -qe '^\[  SKIPPED \]' && res=skip
+	# Report skipped tests as successful since KernelCI doesn't support them
+	echo "$line" | grep -qe '^\[  SKIPPED \]' && res=pass
```
Comment (gctucker) [inline config/rootfs/debos/overlays/libcamera/usr/bin/lc-compliance-parser.sh:37]: Please drop this patch, KernelCI does support skipped tests.

**Review by gctucker — CHANGES_REQUESTED** [review]

**Comment by gctucker** [comment]
Tested OK on staging:
https://staging.kernelci.org/test/job/kernelci/branch/staging-mainline/kernel/staging-mainline-20210611.1/plan/lc-compliance/

**Review by gctucker — APPROVED** [review]

**Pull request merged**